org-code-javabuilder/lib/src/main/java/org/code/javabuilder/CodeExecutionManager.java

```diff
@@ -162,20 +145,29 @@ private void onPreExecute() throws InternalServerException {
     this.overrideOutputStream = new OutputPrintStream(this.outputAdapter);
     System.setOut(this.overrideOutputStream);
     System.setIn(this.overrideInputStream);
+    this.isInitialized = true;
   }
 
   /**
    * Post-execution steps: 1) Notify listeners, 2) clean up global resources, 3) clear temporary
    * folder, 4) close custom in/out streams, 5) Replace System.in/out with original in/out, 6)
    * Destroy Global Protocol
    */
-  private void onPostExecute() throws InternalServerException {
-    if (!this.executionInProgress) {
-      Logger.getLogger(MAIN_LOGGER)
-          .warning("onPostExecute() called while execution not in progress.");
-      return;
+  private void onPostExecute() {
+    // Notify user and listeners
+    try {
+      this.outputAdapter.sendMessage(new StatusMessage(StatusMessageKey.EXITED));
```

Comment:
this logic for catching output adapter messages is repeated in `ExceptionHandler`, would it make sense for this logic to move to `AWSOutputAdapter` instead?

Reply:
Yeah, so I went back and forth on this - unfortunately the situation with the OutputAdapter is a little weird right now (definitely open to suggestions on how to streamline this). During code execution, we do want the output adapter to throw an exception if it encounters a GoneException because that's how we terminate code execution early, and we have logic in the ExceptionHandler to catch it and handle it accordingly. However, outside of code execution (which is here, in LambdaRequestHandler, and in ExceptionHandler), we don't want any exceptions to propagate because that will result in an uncaught exception being thrown from the handler. For that reason, I added these try-catch blocks, but yep that results in this block of code being copied multiple times. 

One way we could avoid copying code might be to move this into a util? something like `ProtocolUtils.safelySendMessage(OutputAdapter, ClientMessage)`?

Reply:
yeah refactoring to a util seems like a good step

Reply:
Cool, I'll make a quick util for this